src/Common/Config/YAMLParser.cpp

```diff
@@ -0,0 +1,196 @@
+            Poco::XML::Element * new_node = xml->createElement(key);
+            processNode(val_node, xml, new_node, ret_vec);
+            for (auto & iter : ret_vec)
+            {
+                new_node->setAttribute(iter.first, iter.second);
+            }
+            ret_vec.clear();
+            xml_node->appendChild(new_node);
+        }
+        else if (val_node.IsSequence())
+        {
+            /// YAML node which contains other nodes and/or values
+            std::string key, value;
+            key = key_node.as<std::string>();
+            /// Firstly we iterate through the sequence to get all xml attributes
+            for (const auto & seq_node : val_node)
+            {
+                /// if val_node is just a simple value node, it cannot contain attribute
+                /// attribute nodes consist of an attribute key and a value
+                /// so, in sequence with nodes ' - value ' and '- "@key":value' we want to check only the second one
+                if (!seq_node.IsScalar())
+                {
+                    seqNodeGetAttrs(seq_node, ret_vec);
+                }
+            }
+            /// Then we iterate through it again to convert all actual values to xml and set their attributes
+            for (const auto & seq_node : val_node)
+            {
+                if (seq_node.IsScalar())
+                {
+                    value = seq_node.as<std::string>();
+                    if (!key.starts_with(YAML_ATTRIBUTE_PREFIX))
+                    {
+                        Poco::XML::Element * new_node = xml->createElement(key);
+                        for (auto & iter : ret_vec)
+                        {
+                            new_node->setAttribute(iter.first, iter.second);
+                        }
+                        Poco::XML::Text * xml_value = xml->createTextNode(value);
+                        new_node->appendChild(xml_value);
+                        xml_node->appendChild(new_node);
+                    }
+                }
+                else
+                {
+                    const auto check_attr = seq_node.begin();
+                    if (check_attr->second.IsScalar())
+                    {
+                        if (check_attr->first.as<std::string>().starts_with(YAML_ATTRIBUTE_PREFIX))
+                        {
+                            continue;
+                        }
+                    }
+                    Poco::XML::Element * new_node = xml->createElement(key);
+                    processNode(seq_node, xml, new_node, ret_vec);
+                    for (auto & iter : ret_vec)
+                    {
+                        new_node->setAttribute(iter.first, iter.second);
+                    }
+                    xml_node->appendChild(new_node);
+                }
+            }
+            ret_vec.clear();
+        }
+        else
+        {
+            /// yaml-cpp cannot define the node type,
+            return false;
```

Comment:
Is this case really possible? If yes could we throw an exception instead of returning false or just ignore this case? Just to make the things easier. I mean this function will throw exception on fail anyway (for example `as<string>()` can throw an exception). And because of that it doesn't look to be good to return false on some other fails sometimes. Usually we either throw exceptions OR return false, but not the both things together.

Reply:
It seems that the only type that can cause that case is NodeType::Undefined. Actually, that means that something is wrong and we cannot parse it correctly, although we managed to YAML::parse the file. For example, we can get node with this type when we try to use operator `['node1']` on a map, which doesn't contain a node with name `node1`. It's unlikely that we will get a node with that type during normal iteration, but we still need to check for it.

Reply:
Then it makes sense to throw an exception with our error code `CANNOT_PARSE_YAML` here instead of returning false.